gsplat/cuda/csrc/helpers.cuh

```diff
@@ -104,36 +110,26 @@ inline __device__ float4 transform_4x4(const float *mat, const float3 p) {
 }
 
 inline __device__ float2 project_pix(
-    const float *mat, const float3 p, const dim3 img_size, const float2 pp
+    const float2 fxfy, const float3 p_view, const float2 pp
```

Comment:
The API of this internal helper is also changed: In the previous version, the parameter was assumed to be a (model-)view-projection matrix instead of a projection matrix, and `p` was in world coordinates. Now the transformation to camera coordinates is handled separately and this method only performs the projection, for better separation of concerns.

Reply:
yes, apologies. for the sake of laziness, we compute the full matrix, proj_mat @ view_mat and pass that matrix around. we pass both this full matrix and view matrix around because both are necessary for the backward pass. this new API works well too.